src/computer.cpp

```diff
@@ -1262,6 +1263,234 @@ SHORTLY. TO ENSURE YOUR SAFETY PLEASE FOLLOW THE STEPS BELOW. \n\
+                                error = true;
+                                options.clear(); // Disable the terminal.
+                                activate_failure( COMPFAIL_SHUTDOWN );
+                                break;
+                            }
+                            g->m.adjust_radiation( dest, rng( 20, 50 ) );
+                            for( const tripoint &radorigin : g->m.points_in_radius( dest, 5 ) ) {
+                                g->m.adjust_radiation( radorigin, rng( 20, 50 ) / ( rl_dist( radorigin,
+                                                       dest ) > 0 ? rl_dist( radorigin, dest ) : 1 ) );
+                            }
+                            // if unshielded, rad source irradiates player directly, reduced by distance to source
+                            if( g->m.pl_sees( dest, 10 ) ) {
+                                g->u.irradiate( rng( 5, 25 ) / rl_dist( g->u.pos(), dest ) );
+                            }
+                        }
+                        if( !error && platform_exists ) {
+                            print_error( _( "PROCESSING...  CYCLE COMPLETE." ) );
+                            print_error( _( "GEIGER COUNTER @ PLATFORM: %s mSv/h." ), g->m.get_radiation( dest ) );
+                        }
+                    }
+                }
+            }
+            if( !platform_exists ) {
+                print_error(
+                    _( "CRITICAL ERROR... RADIATION PLATFORM UNRESPONSIVE.  COMPLY TO PROCEDURE RP_M_01_rev.03." ) );
+            }
+            if( !error ) {
+                query_any( _( "Press any key..." ) );
+            }
+            break;
+        }
+
+        // geiger counter for irradiator, primary measurement at t_rad_platform, secondary at player loacation
+        case COMPACT_GEIGER: {
+            g->u.moves -= 30;
+            tripoint platform;
+            bool source_exists = false;
+            int sum_rads = 0;
+            int peak_rad = 0;
+            int tiles_counted = 0;
+            print_error( _( "RADIATION MEASUREMENTS:" ) );
+            for( const tripoint &dest : g->m.points_in_radius( g->u.pos(), 10 ) ) {
+                if( g->m.ter( dest ) == t_rad_platform ) {
+                    source_exists = true;
+                    platform = dest;
+                }
+            }
+            if( source_exists ) {
+                for( const tripoint &dest : g->m.points_in_radius( platform, 3 ) ) {
+                    sum_rads += g->m.get_radiation( dest );
+                    tiles_counted ++;
+                    if( g->m.get_radiation( dest ) > peak_rad ) {
+                        peak_rad = g->m.get_radiation( dest );
+                    }
+                    sum_rads += g->m.get_radiation( platform );
+                    tiles_counted ++;
+                    if( g->m.get_radiation( platform ) > peak_rad ) {
+                        peak_rad = g->m.get_radiation( dest );
```

Comment:
Shouldn't `platform` be used here instead of `dest`?

Reply:
True. Fixing.